### ISSUE TITLE
Crash on Load of Chamber 8 on Delta Emulator iOS / mupen64plus

### ISSUE THREAD

**Issue by Complexicon** [mwpenny/portal64-still-alive#76]
#### Version of the game
 - Portal64 ROM Built with Build ID 13095453 of Portal on Steam, based on commit
 - [7964ae1](https://github.com/mwpenny/portal64-still-alive/commit/7964ae17db2e29df6cc021086a8f8bff838d01b0)

#### Original Hardware or Emulator?
Ran on Delta Emulator for iOS which is a wrapper for mupen64plus
Also tested on mupen64plus directly on Windows

#### What is the bug?
Full on Freeze no response to controls.

#### Steps to reproduce the behavior:
1. In test chamber 7 
2. finish test chamber
3. enter elevator
4. hangup occurs on load of chamber 8

#### Additional context
if needed i can provide a save state which should include a memory dump etc

**Comment by hackgrid** [comment]
If it does not happen on real hardware, it is the emulators fault. We can
not fix this. Try ares for a good n64 emulator instead.

Complexicon ***@***.***> schrieb am Fr., 14. Juni 2024, 14:11:

**Comment by mwpenny** [comment]
@hackgrid is right that we can't fix emulator bugs, but it's interesting this is a mupen64plus wrapper since I have used both Ares and mupen64plus without issue.

A problem I've encountered before is that the Makefile dependencies aren't entirely correct, meaning when rebuilding after pulling changes some code doesn't get recompiled when it should. With some changes like struct layout this can cause a crash.

If you built the ROM, pulled new changes, and again then try doing a `make clean` followed by `make` and see if that solves the problem. The Makefile issue will eventually be fixed but there is some other work taking priority right now (chamber 15).
